source/reference/verification/embedded.txt

```diff
@@ -0,0 +1,114 @@
+
+.. _c2c-embedded-verifier:
+
+=============================
+Verify with Embedded Verifier
+=============================
+
+.. default-domain:: mongodb
+
+.. facet::
+   :name: genre
+   :values: tutorial
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+``mongosync`` includes an embedded verifier to check the
+migration to the destination cluster. The verifier checks all
+supported collections on the destination cluster to verify a
+successful sync. 
+
+.. important::
+
+   .. include:: /includes/fact-verifier-unsupported
+
+.. versionadded:: 1.9
+
+About this Task
+---------------
+
+.. include:: /includes/fact-verifier-limitations
+
+Steps
+-----
+
+.. procedure::
+   :style: normal
+
+   .. step:: Initialize ``mongosync``
+
+     Initialize the ``mongosync`` process:
+
+     .. code-block:: sh
+
+        ./bin/mongosync \
+            --logPath /var/log/mongosync \
+            --cluster0 "mongodb://clusterAdmin:superSecret@clusterOne01.fancyCorp.com:20020,clusterOne02.fancyCorp.com:20020,clusterOne03.fancyCorp.com:20020" \
+            --cluster1 "mongodb://clusterAdmin:superSecret@clusterTwo01.fancyCorp.com:20020,clusterTwo02.fancyCorp.com:20020,clusterTwo03.fancyCorp.com:20020"
+
+   .. step:: Start the Sync
+
+      To start syncing data from the source cluster to the
+      destination, use the :ref:`/start <c2c-api-start>`
+      endpoint. To use the embedded verifier, set the
+      ``verification.enabled`` field to ``true``.
+
+      .. code-block:: sh
+
+        curl localhost:27182/api/v1/start -XPOST \
+            --data ' {
+                "source": "cluster0",
+                "destination": "cluster1",
+                "verification": {
+                   "enabled": true
+                }
+            } '
+
+      Example output:
+
+      .. literalinclude:: /includes/api/responses/success.json
+         :language: json
+         :copyable: false
+
+   .. step:: Examine Progress
+
+      To examine the status of the sync, use the :ref:`/progress
+      <c2c-api-progress>` endpoint:
+
+      .. literalinclude:: /includes/api/requests/progress.sh
+         :language: shell
+
+      Example output:
+
+      .. literalinclude:: /includes/api/responses/progress.json
+         :language: json
+         :copyable: false
+
+      Examine the ``verification`` response field for
+      information on the status of the embedded verifier.
+
+Behavior
+--------
```

Comment:
In https://jira.mongodb.org/browse/REP-5163 we will add a new check in our handling of the /start endpoint. If verification is enabled, mongosync will check that there is at least 0.5 GB per 1 million docs + 10 GB of memory available. If not, mongosync will error. The user should either increase their memory and restart mongosync or disable verification.

Reply:
Fixed